### ISSUE TITLE
Explain how this compares to Boost's config mechanism

### ISSUE THREAD

**Issue by eyalroz** [xeerx/config#1]
The Boost libraries also have a bunch of headers which determine the compiler, platform, standard library etc. How does this repository compares with boost's mechamism? IMHO that should be explained in the README or in a Wiki page.

**Comment by xeerx** [comment]
I don't know much about `Boost's config mechanism`

But overall, my library provides easy and clean code, easy to understand and modify, as well as simple.

The code doesn't do magic, just basically defines a few things.

In my projects I don't like to include libraries that contain a lot of code and I don't like defining a lot of macros, for example when I include my library it only defines:

- 2 user-defined macros
> enough to control the library

- 1 helper macros
> Sufficient to facilitate the use of the library

- 7 architecture macros
> Provides all the information I need regarding architecture

- 5 compiler macros
> Provides all the information I need regarding compiler

- 5 stdlib macros
> Provides all the information I need regarding stdlib

- 4 platform macros
> Provides all the information I need regarding platform

- 17 attribute macros
> Provides all the common attributes/features that I usually use in my projects

**in general i will got 41 macro, that enough to my projects.**


Also when I looked at the source code for `Boost config`, I found that it does lots and lots of old compilers version checks , which we don't need today (I don't know anyone who still uses `GCC v(2-8)` while At the moment there is version `12`!

In short, the library doesn't do a lot of work, it just provides basic project requirements that we all need on a regular basis.

**Comment by eyalroz** [comment]
> (I don't know anyone who still uses GCC v(2-8) 

Not sure what you're talking about, GCC 8 was just released in 2018, that's practically yesterday...

And millions of systems around the world actually use much older compilers.

Anyway, what you've said here in the comment should probably be explained in the README.md.
